packages/js/product-editor/src/hooks/use-meta-entity-prop.ts

```diff
@@ -0,0 +1,50 @@
+/**
+ * External dependencies
+ */
+import { useEntityProp } from '@wordpress/core-data';
+
+interface Metadata {
+	id?: number;
+	key: string;
+	value: string;
+}
+
+function useMetaEntityProp(
```

Comment:
Love, love, love the idea of this hook... I actually had on my list for later today to create an issue for such a hook, since we have so many places in our blocks where we are using`useEntityProp<>( 'postType, 'product', 'foo )` and it seems silly to have to repeat `postType`, and `product` all over the place...

I think we should name this hook more specifically, since it is tied to products... `useProductEntityProp()`

And, as mentioned above, I think we can handle both regular properties and meta properties without an `isMeta` flag...

If you look at `property` and it starts with `meta_data.`, you would handle it as a meta property. Otherwise, it is a normal property.

Btw... such a hook should have some tests for it! 😄